Add synthetic mode

nikhilwoodruff opened the pull request:
# New feature: Synthetic mode for the UK system

This adds a way for everyone who doesn't have access to the UK microdata to run the server - it should work as described in the README (just changing a bool option from false to true).

MaxGhenis commented:
Could this be a flag passed to `make debug-server` instead of requiring code modification?

nikhilwoodruff commented:
@MaxGhenis good idea, switched to that. The flag is `POLICYENGINE_SYNTH_UK=1 make debug-server`, plus I added a shortcut in `make debug-server-synth`.
